sanity/loop.go

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package sanity
+
+import (
+	"fmt"
+	"syscall"
+)
+
+func init() {
+	checks = append(checks, checkLoopControl)
+}
+
+var (
+	loopControlPath  = "/dev/loop-control"
+	loopControlMajor = 10
+	loopControlMinor = 237
+)
+
+var majorMinor = func(rdev int) (major, minor int) {
+	major = int(rdev / 256)
+	minor = int(rdev % 256)
+	return major, minor
+}
+
+func checkLoopControl() error {
+	var st syscall.Stat_t
+	err := syscall.Stat(loopControlPath, &st)
```

Comment:
As discussed on IRC, this should depend on `squashfs.FsType()`, we only need loop when the actual `squashfs` is in use.

Maybe it'd make sense to move this check into the the squashfs one?

Reply:
Done.